### PR TITLE
upgrade some dependencies

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://github.com/DataBiosphere/leonardo/issues/732

akka-http `10.1.5` had some [security fix](https://akka.io/blog/news/2018/09/06/akka-http-10.1.5-10.0.14-security-fix-released)

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -96,7 +95,6 @@ object Dependencies {
     scalaLogging,
     swaggerUi,
     ficus,
-    cats,
```
Comment (MatthewBemis) [inline project/Dependencies.scala]: is this getting pulled in through some other dep now?

Comment (Qi77Qi) [reply project/Dependencies.scala]: yea..`workbench-utils` pulls this in transitively now

**Comment by Qi77Qi** [comment]
jenkins retest

File: src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutesSpec.scala [hunk 2]
```diff
@@ -135,7 +135,7 @@ class ProxyRoutesSpec extends FlatSpec with BeforeAndAfterAll with BeforeAndAfte
       Get(s"/$prefix/$googleProject/$clusterName").addHeader(Cookie(tokenCookie))
         .addHeader(RawHeader("foo", "bar"))
         .addHeader(RawHeader("baz", "biz")) ~> leoRoutes.route ~> check {
-        responseAs[Data].headers should contain allElementsOf Map("foo" -> "bar", "baz" -> "biz")
+        responseAs[Data].headers.toList should contain allElementsOf Map("foo" -> "bar", "baz" -> "biz").toList
```
Comment (Qi77Qi) [inline src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutesSpec.scala]: this is due to https://github.com/scalatest/scalatest/issues/1224. Basically with `partial-unification` on, `contain` doesn't work for `Map[A, B]`

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala [hunk 3]
```diff
@@ -357,12 +357,10 @@ class ClusterMonitorActor(val cluster: Cluster,
   private def getClusterInstances: Future[Set[Instance]] = {
     for {
       map <- gdDAO.getClusterInstances(cluster.googleProject, cluster.clusterName)
-      instances <- Future.traverse(map) { case (role, instances) =>
-        Future.traverse(instances) { instance =>
-          googleComputeDAO.getInstance(instance).map(_.map(_.copy(dataprocRole = Some(role))))
-        }
+      instances <- map.toList.flatTraverse { case (role, instances) =>
```
Comment (Qi77Qi) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala]: `Future.traverse` doesn't work for `Map` with `partial-unification` on. Hence refactor a bit

File: project/Settings.scala [hunk 4]
```diff
@@ -22,11 +22,51 @@ object Settings {
...
+    "-deprecation",                      // Emit warning and location for usages of deprecated APIs.
+    "-encoding", "utf-8",                // Specify character encoding used by source files.
+    "-explaintypes",                     // Explain type errors in more detail.
+    "-feature",                          // Emit warning and location for usages of features that should be imported explicitly.
+    "-language:existentials",            // Existential types (besides wildcard types) can be written and inferred
+    "-language:higherKinds",             // Allow higher-kinded types
+    "-language:implicitConversions",     // Allow definition of implicit functions called views
+    "-unchecked",                        // Enable additional warnings where generated code depends on assumptions.
+    "-Xcheckinit",                       // Wrap field accessors to throw an exception on uninitialized access.
+    "-Xfatal-warnings",                  // Fail the compilation if there are any warnings.
```
Comment (MatthewBemis) [inline project/Settings.scala]: when it says _any_ warning does it really mean _any_ warning? i.e. all of the additional warnings that are being turned on in this PR will be grounds to fail the compilation?

**Comment by MatthewBemis** [comment]
also, one other thing: it looks like the leo build is still failing even after the retest

File: project/Settings.scala (hunk 4 above)
Comment (Qi77Qi) [reply project/Settings.scala]: yep..with this option on, compilation will fail when there's warning.

Comment (MatthewBemis) [reply project/Settings.scala]: i think this is a good idea but we should communicate it to the team as it's a behavioral change that developers are likely to encounter

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
time out on 
```
Cromiam 500 {'ok': False, 'systems': {'Cromwell': {'messages': ['<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\n<html><head>\n<title>503 Service Unavailable</title>\n</head><body>\n<h1>Service Unavailable</h1>\n<p>The server is temporarily unable to service your\nrequest due to maintenance downtime or capacity\nproblems. Please try again later.</p>\n<hr>\n<address>Apache Server at cromwell-fiab.dsde-qa.broadinstitute.org Port 443</address>\n</body></html>\n'], 'ok': False}, 'Sam': {'ok': True}}}
  Cromiam                              False
JobManager 503 {'detail': {'available': False}, 'status': 503, 'title': 'Service Unavailable', 'type': 'about:blank'}
  JobManager                           False
----------------------------------------------
  FiaB ready                           False
```

jenkins retest

**Comment by Qi77Qi** [comment]
I ended up not upgrading akka-http because newer akka-http version seems invalidate `sun.net.spi.nameservice.provider.n` setting somehow

**Pull request merged**